oauth/lib/redirecturiparser.cpp

```diff
@@ -0,0 +1,51 @@
+/****************************************************************************
+**
+** Copyright (C) 2021 Open Mobile Platform LLC
+**
+****************************************************************************/
+
+#include "redirecturiparser_p.h"
+
+#include <QtCore/QUrl>
+#include <QtCore/QUrlQuery>
+#include <QtCore/QStringList>
+#include <QtCore/QString>
+#include <QtCore/QByteArray>
+#include <QtCore/QPair>
+#include <QtCore/QList>
+
+QVariantMap SailfishOS::OAuth::parseRedirectUri(const QString &redirectUri, QString *errorMessage)
+{
+    QVariantMap data;
+
+    const QUrl url(redirectUri);
+    const QUrlQuery query(url.query());
+    const QString fragment = url.fragment(QUrl::FullyEncoded);
+    const QStringList fragments = fragment.split(QChar('&'));
+
+    QString errorString;
+
+    QList<QPair<QString, QString> > fragmentPairs;
+    for (const QString &f : fragments) {
```

Comment:
Could you rename please 'f' to 'fragment' or something more clear? (if the line 23 is unused)

Reply:
Done